nhrpd/nhrp_vty.c

```diff
@@ -741,8 +741,8 @@ static void show_ip_nhrp_shortcut(struct nhrp_shortcut *s, void *pctx)
 	ctx->count++;
 
 	c = s->cache;
-	if (c)
-		sockunion2str(&c->remote_addr, buf2, sizeof(buf2));
+	if (!c || !sockunion2str(&c->remote_addr, buf2, sizeof(buf2)))
```

Comment:
I would just set buf2[0] = `\0`; immediately after line 743. and we won't have this awkward if statement.

Reply:
That relies on the assumption that buf2 would not be altered by sockunion2str in case that function fails. I am not sure that is a safe assumption?

Reply:
- this should be done since inet_ntop may fail.
- as a separate subject, I wonder if %pSU is safe in case sockunion2str fails.

Reply:
I had a look, and %pSU is not taking any precautions for failed inet_ntop. Although I think it is impossible for that function to actually fail, since I see no way for the inputs to be invalid.

Reply:
so, we do not need to check return status of sockunion2str.
the recommendation frmo Donale is here as an additional sanity check ( SA warning?)

Reply:
This change is done now

Reply:
@donaldsharp. Is there any reason why this pull request can't be merged?